### PR TITLE
Migrate GitHub actions from AdoptOpenJDK to Eclipse Temurin [HZ-4195]

### PR DESCRIPTION
**Pull request by JackPGreen**
The `setup-java` [documentation highlights that AdoptOpenJDK no longer receives security updates](https://github.com/actions/setup-java#:~:text=NOTE%3A%20AdoptOpenJDK%20got%20moved%20to%20Eclipse%20Temurin), and instead the Eclipse Temurin JDK should be used.

Fixes [HZ-4195](https://hazelcast.atlassian.net/browse/HZ-4195)

[HZ-4195]: https://hazelcast.atlassian.net/browse/HZ-4195?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by kwart — DISMISSED** [review]

File: .github/workflows/coverage_runner.yaml
Comment (ldziedziul) [inline .github/workflows/coverage_runner.yaml:45]: What about java version? Maybe we can update it?

Comment (JackPGreen) [reply .github/workflows/coverage_runner.yaml:45]: > What about java version? Maybe we can update it?

I think we _can_ upgrade it, but I'm not sure we need to. The [documentation says that Java 8 is ok for the integration tests](https://github.com/hazelcast/hazelcast-go-client?tab=readme-ov-file#running-the-tests).

**Comment by JackPGreen** [comment]
@yuce are you able to help get this PR merged, please?

It's stuck waiting for the `test` job to complete but I can't see where that is (i.e. GitHub action or Jenkins).

File: .github/workflows/coverage_runner.yaml
Comment (yuce) [reply .github/workflows/coverage_runner.yaml:45]: I think it would be better to set it to the minimal version required by Hazelcast.

**Comment by yuce** [comment]
@JackPGreen The min reviewer count was set to 2 for this repo. The tests will start once everyone approves it.

File: .github/workflows/coverage_runner.yaml
Comment (JackPGreen) [reply .github/workflows/coverage_runner.yaml]: > I think it would be better to set it to the minimal version required by Hazelcast.

e09a31ecdf176fee7d7363c4ebb894fd217bab64

**Review by yuce — APPROVED** [review]

**Review by srknzl — APPROVED** [review]

**Comment by JackPGreen** [comment]
> > @yuce are you able to help get this PR merged, please?
> > 
> > It's stuck waiting for the `test` job to complete but I can't see where that is (i.e. GitHub action or Jenkins).
>
> @JackPGreen The min reviewer count was set to 2 for this repo. The tests will start once everyone approves it.

@yuce thanks, there are two approvers but I still have:
> test Expected — Waiting for status to be reported

Do you know where this job is triggered, please?

**Comment by yuce** [comment]
That seems like a problem with GH, all tests passed.
I'll merge the PR.

**Pull request merged**
